src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/NuGetRestorer.cs

```diff
@@ -0,0 +1,140 @@
+
+            // when TargetFrameworks or TargetFrameworkMoniker changes, reset subscriptions so that
+            // any new configured projects are picked up
+            if (HasTargetFrameworkChanged(update))
+            {
+                await ResetSubscriptions().ConfigureAwait(false);
+            }
+        }
+
+        protected override async Task InitializeCoreAsync(CancellationToken cancellationToken)
+        {            
+            await ResetSubscriptions().ConfigureAwait(false);
+        }
+
+        protected override Task DisposeCoreAsync(bool initialized)
+        {
+            _evaluationSubscriptionLink?.Dispose();
+            _targetFrameworkSubscriptionLink?.Dispose();
+            return Task.CompletedTask;
+        }
+
+        private async Task ResetSubscriptions()
+        {
+            _evaluationSubscriptionLink?.Dispose();
+
+            var currentProjects = await _activeConfiguredProjectsProvider.GetActiveConfiguredProjectsAsync().ConfigureAwait(false);
+
+            if (currentProjects.Any())
+            {
+                var sourceLinkOptions = new StandardRuleDataflowLinkOptions
+                {
+                    RuleNames = _evaluationWatchedRules,
+                    PropagateCompletion = true
+                };
+
+                var sourceBlocks = currentProjects.Select(
+                    cp => cp.Services.ProjectSubscription.ProjectRuleSource.SourceBlock.SyncLinkOptions<IProjectValueVersions>(sourceLinkOptions));
+
+                var target = new ActionBlock<Tuple<ImmutableList<IProjectValueVersions>, TIdentityDictionary>>(ProjectPropertyChangedAsync);
+
+                var targetLinkOptions = new DataflowLinkOptions { PropagateCompletion = true };
+
+                _evaluationSubscriptionLink = ProjectDataSources.SyncLinkTo(sourceBlocks.ToImmutableList(), target, targetLinkOptions);
+            }
+        }
+
+        private async Task ProjectPropertyChangedAsync(Tuple<ImmutableList<IProjectValueVersions>, TIdentityDictionary> sources)
+        {
+            IVsProjectRestoreInfo projectRestoreInfo = ProjectRestoreInfoBuilder.Build(sources.Item1);
+
+            if (projectRestoreInfo != null)
+            {
+                await _solutionRestoreService
+                    .NominateProjectAsync(_projectVsServices.Project.FullPath, projectRestoreInfo, CancellationToken.None)
+                    .ConfigureAwait(false);
+            }
+        }
+
+        private static bool HasTargetFrameworkChanged(IProjectVersionedValue<IProjectSubscriptionUpdate> update)
+        {
+            IProjectChangeDescription projectChange;
+            if (update.Value.ProjectChanges.TryGetValue(ConfigurationGeneral.SchemaName, out projectChange))
+            {
+                var changedProperties = projectChange.Difference.ChangedProperties;
+                return changedProperties.Contains(ConfigurationGeneral.TargetFrameworksProperty)
```

Comment:
Again, don't think you should be looking for TargetFrameworks property - it's irrelevant to you, you care about then configs change regardless of whether they are backed by target frameworks.